oidc-client-app-samples/android-client-app-sample/app/src/main/java/com/wso2is/androidsample/openid/LogoutRequest.java

```diff
@@ -0,0 +1,94 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.wso2is.androidsample.openid;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.support.customtabs.CustomTabsIntent;
+import com.wso2is.androidsample.mgt.AuthStateManager;
+import com.wso2is.androidsample.mgt.ConfigManager;
+import net.openid.appauth.AuthState;
+
+import java.lang.ref.WeakReference;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static com.wso2is.androidsample.activities.UserActivity.idToken;
+import static com.wso2is.androidsample.activities.UserActivity.state;
+
+/**
+ * This class facilitates the logout function of the application.
+ * User's state will be reset and will be logged out from the WSO2 IS.
+ */
+public class LogoutRequest {
+
+    private static final AtomicReference<WeakReference<LogoutRequest>> instance = new AtomicReference<>(new WeakReference<LogoutRequest>(null));
+
+    private LogoutRequest() {
+
+    }
+
+    /**
+     * Returns an instance of the LogoutRequest class.
+     *
+     * @return LogoutRequest instance.
+     */
+    public static LogoutRequest getInstance() {
+
+        LogoutRequest logoutRequest = instance.get().get();
+        if (logoutRequest == null) {
+            logoutRequest = new LogoutRequest();
+            instance.set(new WeakReference<>(logoutRequest));
+        }
+        return logoutRequest;
+    }
+
+    /**
+     * The user will be signed out from the application and the WSO2 IS.
+     *
+     * @param context Current context of the application.
+     */
+    public void signOut(Context context) {
+
+        AuthStateManager authStateManager = AuthStateManager.getInstance(context);
+        ConfigManager configuration = ConfigManager.getInstance(context);
+
+        // discarding the authorization and token state, but retaining the configuration
+        // to save from retrieving them again.
+        AuthState currentState = authStateManager.getCurrentState();
+        AuthState clearedState = new AuthState(currentState.getAuthorizationServiceConfiguration());
+
+        if (currentState.getLastRegistrationResponse() != null) {
+            clearedState.update(currentState.getLastRegistrationResponse());
+        }
+
+        authStateManager.replaceState(clearedState);
+
+        String logout_uri = configuration.getLogoutEndpointUri().toString();
+        String redirect = configuration.getRedirectUri().toString();
+        String url = logout_uri + "?id_token_hint=" + idToken + "&post_logout_redirect_uri=" + redirect
```

Comment:
Shall we use string builder instead

Reply:
Fixed in PR https://github.com/wso2/samples-is/pull/174.